catalog/resource_external_location.go

```diff
@@ -2,45 +2,36 @@ package catalog
 
 import (
 	"context"
-	"net/url"
 
+	"github.com/databricks/databricks-sdk-go/service/catalog"
 	"github.com/databricks/terraform-provider-databricks/common"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
-type ExternalLocationsAPI struct {
-	client  *common.DatabricksClient
-	context context.Context
-}
-
-func NewExternalLocationsAPI(ctx context.Context, m any) ExternalLocationsAPI {
-	return ExternalLocationsAPI{m.(*common.DatabricksClient), context.WithValue(ctx, common.Api, common.API_2_1)}
-}
-
-type ExternalLocationInfo struct {
-	Name           string `json:"name" tf:"force_new"`
-	URL            string `json:"url"`
-	CredentialName string `json:"credential_name"`
-	Comment        string `json:"comment,omitempty"`
-	SkipValidation bool   `json:"skip_validation,omitempty"`
-	Owner          string `json:"owner,omitempty" tf:"computed"`
-	MetastoreID    string `json:"metastore_id,omitempty" tf:"computed"`
-	ReadOnly       bool   `json:"read_only,omitempty"`
-}
+// This structure contains the fields of both catalog.UpdateExternalLocation and catalog.CreateExternalLocation
 
-func (a ExternalLocationsAPI) create(el *ExternalLocationInfo) error {
-	return a.client.Post(a.context, "/unity-catalog/external-locations", el, &el)
+type SseEncryptionDetails struct {
+	// algorithm enum values: "AWS_SSE_S3", "AWS_SSE_KMS"
+	Algorithm    string `json:"algorithm,omitempty"`
+	AwsKmsKeyArn string `json:"aws_kms_key_arn,omitempty"`
 }
 
-func (a ExternalLocationsAPI) get(name string) (el ExternalLocationInfo, err error) {
-	err = a.client.Get(a.context, "/unity-catalog/external-locations/"+url.PathEscape(name), nil, &el)
-	return
+type EncryptionDetails struct {
+	// This is a oneOf type, but only one type defined currently:
+	SseEncDetails *SseEncryptionDetails `json:"sse_encryption_details,omitempty"`
 }
```

Comment:
Should use the Go SDK types, right?  See [this PR](https://github.com/databricks/terraform-provider-databricks/pull/2539/files#diff-0920ed275cfdb0ac2f823d9380fcfe2cbba9c0622c8f487836f1f79a6f5a06c5).